pkg/filesystem/fs.go

```diff
@@ -143,14 +143,41 @@ func NewFileSystem(ctx context.Context, opt ...NewFSOpt) (*Filesystem, error) {
 			return nil
 		})
 	}
-	if err := eg.Wait(); err != nil {
+	if err := egRecover.Wait(); err != nil {
 		return nil, err
 	}
 
-	for _, d := range liveDaemons {
-		fs.TryRetainSharedDaemon(d)
+	newNydusImageBinaryCommit, err := daemon.GetDaemonGitCommit(fs.nydusImageBinaryPath)
```

Comment:
Should be `nydusdBinaryPath`, `newNydusdBinaryCommit`.

Reply:
done.